ruby-challenges.rb

```diff
@@ -10,22 +16,91 @@
 letter_t = 't'
 # Expected output: ['tea', 'water', 'soda water']
 
+def has_o(array,str)                   #declare function that takes an array and string as args
+    array.filter do |value|             #filter the array for the values    
+        value.include?(str)             #if the values inlcude the given string then they get filtered
+    end
 
+end
+p has_o(beverages_array, letter_o)
+p has_o(beverages_array, letter_t)
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
+# Pseudo code
+# Input:takes in a hash 
+# Transformation: get rid of nested arrays, turn hash into an array and sort in alphabetical order.
+#first with the fresh variable able to use the flatten method twice to un nest the arrays
+#declare empty array final
+#then map the array if the value is a string it is pushed onto the empty array
+#use the sort method to put them in abc order.
 
+# Output:an array that is in alphabeticalorder
 us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Vermont'] }
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Vermont', 'Washington'] 
 
+def usa(hash)
+    fresh = hash.flatten.flatten
+    final = []
+    fresh.map do |value|
```

Comment:
Since map likes to return arrays. What is a refactor you can use to take advantage of that feature?